ros_ws/src/autonomous/emergency_stop/src/emergency_stop.cpp

```diff
@@ -2,54 +2,114 @@
 #include <std_msgs/Time.h>
 
 EmergencyStop::EmergencyStop()
+    : m_emergency_status(EmergencyStatus::UNUSED)
+    , m_debug_geometry(this->m_node_handle, TOPIC_VISUALIZATION, LIDAR_FRAME)
 {
-    m_lidar_subscriber =
-        m_node_handle.subscribe<sensor_msgs::LaserScan>(TOPIC_LASER_SCAN, 1, &EmergencyStop::lidarCallback, this);
-    m_emergency_stop_publisher = m_node_handle.advertise<std_msgs::Time>(TOPIC_EMERGENCY_STOP, 1);
+    this->m_lidar_subscriber =
+        this->m_node_handle.subscribe<sensor_msgs::LaserScan>(TOPIC_LASER_SCAN, 1, &EmergencyStop::lidarCallback, this);
+    this->m_emergency_stop_publisher = this->m_node_handle.advertise<std_msgs::Time>(TOPIC_EMERGENCY_STOP, 1);
+    this->configureParameters();
+}
+
+void EmergencyStop::configureParameters()
+{
+    ros::NodeHandle private_node_handle("~");
+
+    private_node_handle.getParam(RANGE_THRESHOLD, this->m_range_threshold);
+    if ((this->m_range_threshold > 2.f) || (this->m_range_threshold < 0.3f))
+    {
+        ROS_WARN_STREAM("m_range_threshold is set to "
+                        << this->m_range_threshold << ". This value is currently limited to (0.3, 2). Using default: "
+                        << RANGE_THRESHOLD_DEFAULT << ".");
+        this->m_range_threshold = RANGE_THRESHOLD_DEFAULT;
+    }
+
+    private_node_handle.getParam(MAX_RANGE, this->m_max_range);
+    if ((this->m_max_range > 30.f) || (this->m_max_range <= 2.f))
```

Comment:
If 2 should be included and 30 excluded, then the `>` should be `>=` and vice versa.
```suggestion
    if ((this->m_max_range >= 30.f) || (this->m_max_range < 2.f))
```

Reply:
Yes. After looking over it again, i think it should also be fine to include 30 since this value was our default max value before.